src/pages/sourceSettings/sourceSettings.tsx

```diff
@@ -20,41 +20,72 @@ import Header from './header';
 import { NoMatchFoundState } from './noMatchFoundState';
 import NoSourcesState from './noSourcesState';
 import RowCell from './rowCell';
+import SourcePagination from './sourcePagination';
 import { styles } from './sourceSettings.styles';
 import SourceTable from './sourceTable';
 
 interface Props extends InjectedTranslateProps {
   sources: Provider[];
   error: AxiosError;
   status: FetchStatus;
+  updateFilter: typeof sourcesActions.updateFilterToolbar;
   fetch: typeof sourcesActions.fetchSources;
   remove: typeof sourcesActions.removeSource;
   showDeleteDialog: typeof deleteDialogActions.openModal;
   onAdd: typeof onboardingActions.openModal;
+  pagination: any;
+  query: any;
+  currentFilterType: string;
+  currentFilterValue: string;
 }
 
 interface State {
   selected: string[];
   expanded: string[];
-  query: string;
 }
 
 class SourceSettings extends React.Component<Props, State> {
   constructor(props) {
     super(props);
-    this.state = { selected: [], expanded: [], query: '' };
+    this.state = {
+      selected: [],
+      expanded: [],
+    };
     this.onSelect = this.onSelect.bind(this);
     this.onCollapse = this.onCollapse.bind(this);
     this.onSearch = this.onSearch.bind(this);
+    this.onUpdateFilter = this.onUpdateFilter.bind(this);
   }
 
   public componentDidMount() {
-    this.props.fetch(this.state.query);
+    this.props.fetch();
   }
 
-  public onSearch(query) {
-    this.setState({ query });
-    this.props.fetch(query);
+  public onUpdateFilter(selected: string) {
+    let key = null;
+    if (selected === 'type') {
+      key = 'currentFilterType';
+    }
+    if (selected === 'value') {
+      key = 'currentFilterValue';
+    }
+    return (value: string) => {
+      if (key === null) {
+        return;
+      }
+      this.props.updateFilter({ [key]: value });
+    };
+  }
+
+  public onSearch(searchQuery) {
+    const newQuery = { ...this.props.query, ...searchQuery };
+    const res = Object.keys(newQuery)
+      .filter(k => newQuery[k])
+      .reduce((acc, curr) => {
+        const currQuery = `${curr}=${newQuery[curr]}`;
+        return acc === null ? currQuery : `${acc}&${currQuery}`;
+      }, null);
+    this.props.fetch(res);
```

Comment:
Is there any help text to suggest that the filter must be an exact match?

When filtering by type, I expect to see a filter results row and filter chips, similar to the details pages. 

I also expect this to be a partial, logical AND filter. For example, users should be able to enter "Red" and see all partial matches. However, I only see an unexpected error.

![Screen Shot 2019-05-06 at 11 25 54 AM](https://user-images.githubusercontent.com/17481322/57235816-bbcbae00-6ff1-11e9-8440-2f836cdd76bc.png)

FYI, I cannot get out of this state, unless I delete the filter text and hit enter.

Reply:
This addresses only #694, I can add filtering but the amount of files and code to review will be much bigger.

If that's not a problem on your side, let me know so I will continue working on this.

Reply:
In addition, the error you get is irrelevant to `exact match`. 
The backend will throw an error if the type filter is not one of ["OCP", "AWS'].
I suggest to replace it to selector when `Type`

Reply:
We can merge as is and address #649 separately. I was just confused by the filtering changes here.